OpenRA.Mods.Common/Traits/Turreted.cs

```diff
@@ -85,32 +86,30 @@ public static Func<int> TurretFacingFromInit(IActorInitializer init, int def, st
 		}
 
 		public Turreted(ActorInitializer init, TurretedInfo info)
+			: base(info)
 		{
 			this.info = info;
 			TurretFacing = TurretFacingFromInit(init, info.InitialFacing, info.Turret)();
 		}
 
 		public void Created(Actor self)
 		{
-			attack = self.TraitOrDefault<AttackTurreted>();
+			attack = self.TraitsImplementing<AttackTurreted>().ToArray();
```

Comment:
These two lines may stay the same, but there are others that need to change.  There is currently no way to define the relationship between AttackTurreted and turrets.  Each AttackTurreted will try to control all available turrets.